pennylane/circuit_drawer/mpldrawer.py

```diff
@@ -0,0 +1,657 @@
+    to a value of ``2`` in order to draw it *on top* of the control wires, instead of below them.
+    """
+
+    def __init__(self, n_layers, n_wires, wire_kwargs=None, figsize=None):
+
+        self.n_layers = n_layers
+        self.n_wires = n_wires
+
+        # half the width of a box
+        # is difference between center and edge
+        self._box_dx = 0.4
+
+        self._circ_rad = 0.3
+        self._ctrl_rad = 0.1
+        self._octrl_rad = 0.15
+        self._swap_dx = 0.2
+
+        ## Creating figure and ax
+
+        if figsize is None:
+            figsize = (self.n_layers + 3, self.n_wires + 1)
+
+        self.fig = plt.figure(figsize=figsize)
+        self.ax = self.fig.add_axes(
+            [0, 0, 1, 1],
+            xlim=(-2, self.n_layers + 1),
+            ylim=(-1, self.n_wires),
+            xticks=[],
+            yticks=[],
+        )
+
+        self.ax.invert_yaxis()
+
+        if wire_kwargs is None:
+            wire_kwargs = {}
+
+        # adding wire lines
+        for wire in range(self.n_wires):
+            line = plt.Line2D((-1, self.n_layers), (wire, wire), zorder=1, **wire_kwargs)
+            self.ax.add_line(line)
+
+    def label(self, labels, text_kwargs=None):
+        """Label each wire.
+
+        Args:
+            labels [Iterable[str]]: Iterable of labels for the wires
+
+        Keyword Args:
+            text_kwargs (dict): any matplotlib keywords for a text object, such as font or size.
+
+        **Example**
+
+        .. code-block:: python
+
+            drawer = MPLDrawer(n_wires=2, n_layers=1)
+            drawer.label(["a", "b"])
+
+        .. figure:: ../../_static/drawer/labels.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+        You can also pass any
+        `Matplotlib Text keywords <https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.text.html>`_
+        as a dictionary to the ``text_kwargs`` keyword:
+
+        .. code-block:: python
+
+            drawer = MPLDrawer(n_wires=2, n_layers=1)
+            drawer.label(["a", "b"], text_kwargs={"color": "indigo", "fontsize": "xx-large"})
+
+        .. figure:: ../../_static/drawer/labels_formatted.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+        """
+        if text_kwargs is None:
+            text_kwargs = {}
+
+        for wire, ii_label in enumerate(labels):
+            self.ax.text(-1.5, wire, ii_label, **text_kwargs)
+
+    def box_gate(
+        self, layer, wires, text="", extra_width=0, zorder=0, box_kwargs=None, text_kwargs=None
+    ):
+        """Draws a box and adds label text to its center.
+
+        Args:
+            layer (Int): x coordinate for the box center
+            wires (Union[Int, Iterable[Int]]): y locations to include inside the box.  Only min and max
+                of an Iterable affect the output
+            text (str): string to print at the box's center
+
+        Kwargs:
```

Comment:
The kwargs are rendering in the documentation as a single block of text without spacing; not sure what the issue is, the formatting here looks fine. Maybe it need to be this, to match the others? 

```suggestion
        Keyword Args:
```